ocrd_utils/ocrd_utils/os.py

```diff
@@ -90,15 +94,36 @@ def list_all_resources(executable):
     processor_path_var = '%s_PATH' % executable.replace('-', '_').upper()
     if processor_path_var in environ:
         for processor_path in environ[processor_path_var].split(':'):
-            if isdir(processor_path):
-                candidates += list(scandir(processor_path))
-    datadir = join(XDG_DATA_HOME, 'ocrd-resources', executable)
-    if isdir(datadir):
-        candidates += list(scandir(datadir))
-    systemdir = join('/usr/local/share/ocrd-resources', executable)
-    if isdir(systemdir):
-        candidates += list(scandir(systemdir))
-    return [x.path for x in candidates]
+            if Path(processor_path).is_dir():
+                candidates += Path(processor_path).iterdir()
+    datadir = Path(XDG_DATA_HOME, 'ocrd-resources', executable)
+    if datadir.is_dir():
+        candidates += datadir.iterdir()
+    systemdir = Path('/usr/local/share/ocrd-resources', executable)
+    if systemdir.is_dir():
+        candidates += systemdir.iterdir()
+    # recurse once
```

Comment:
Why? (I don't get why this comment is useful, but this might be just me not understanding the code in depth.)

Reply:
I always find it useful to comment where recursion is happening because that is often a source of unexpected behavior.

In this case, we recurse only one level to support tesseract-style `scripts/Fraktur.traineddata` subdirectories. If we later decide we need full recursion, this is the place to adapt.

Reply:
Ah, okay. Do you think this is a kind of implicit knowledge that should be added?

Reply:
Sure, feel free to propose a better wording.